### ISSUE TITLE
zooming in page curl android

### ISSUE THREAD

**Issue by vanireddy** [harism/android-pagecurl#12]
Hi harism ,
How to implement  zooming feature in android..Because i  am implementing a magazine reading effect

**Comment by harism** [comment]
This kind of functionality goes a bit over the scope of this project but as a start you can try using negative margin values to catch up with the idea on zooming. And for panning you could try to altering CurlRenderer.mPageRectLeft and mPageRectRight (these are responsible for storing margins too).

**Comment by musashi076** [comment]
so in panning, i should change it on the updatePageRects?

i applied it like this:

in CurlView, i add a zoom flag which will be change on the menu of curlactivity. if the menu is on zoom, this will enable the zoom flag and disable all the curl animation in the ontouch of curlview.

sorry im just new in open gl

**Comment by bnaee** [comment]
how should i zoom the page??

**Comment by SolimarFerreira** [comment]
Hi harism, sorry to came back at this old post, but i´m doing aproject using your lib but i need have zoom at the page, at nowadays has support to zoom or any solution ?

thanks

**Comment by eltonpiko** [comment]
Hi all did anyone ever found a solution to implement zooming with page curl lib?

**Comment by SolimarFerreira** [comment]
No...

**Comment by reefdev** [comment]
Have anyone come to a solution at this date about zooming imageview with harism page curl ?

**Comment by syamna** [comment]
Is there anyone find solution for zooming??...I need help for zooming feature

**Comment by eltonpiko** [comment]
no didn't get a solution but got a work around that by tapping on the image in page curl the image open's up in image viewer for zooming.

**Comment by syamna** [comment]
can you share that work around code with me

**Comment by syamna** [comment]
Hi

Thanks for your abrupt reply

Can you please share that work around code with me as today is my dead line
..:(

Syeda Amna Akbar

**Comment by soualid** [comment]
I just managed to achieve this by loading the same bitmap in a hidden `ImageView` over the `CurlView` during `updatePage`. Once it is done, just show it when a scale gesture is detected by a `ScaleGestureDetector`, and hide it when scale factor return to 1.

I used https://github.com/MikeOrtiz/TouchImageView to implement the `TouchImageView`.

**Comment by androidvirus** [comment]
i don't understand the hows it is possible in page curl with zoom function because the page code is on based on x and y co-ordinates and i need the help for sort out it.

**Comment by SmartDroidDeveloper** [comment]
i am working on page curl effect with zoom function but in that page curl, its totally calculated on x and y co-ordinates to manage touch of parent. so the zoom function not work with page curl effect to move the other page and also the large bitmap load the more time to goes the next page with effect. so Is there anyone find the solution for it?

**Comment by naveedahmad99** [comment]
Same issue on my side, So I suggest the imageview onClick on that CurlPage, where user can zoom in or out using ToauchImageView or GestureImageView Libraries. if I implement that then I will share with you the code.

**Comment by ravikumar1111** [comment]
Hi, Could you please share code for Page Curl  zooming feature.

**Comment by umitgoren** [comment]
Here's the zoomable version: https://play.google.com/store/apps/details?id=net.codecanyon.trimax.flipbook

**Comment by mayurprasad** [comment]
Hey umitgoren is there any source code or any reference code of your shared link?

**Comment by umitgoren** [comment]
@mayurprasad The link will be on google play store page in a week or so.

**Comment by Canibrahim** [comment]
@umitgoren kodu satıyormusunuz ?

**Comment by umitgoren** [comment]
Evet: https://codecanyon.net/item/android-flipbook-pdf-generator/15691959
